From 39f0b1a5f40d62eccde21402f0f950ac8c0344f0 Mon Sep 17 00:00:00 2001
From: Hugo Villeneuve <hvilleneuve@dimonoff.com>
Date: Thu, 18 Apr 2024 14:23:13 -0400
Subject: [PATCH 1/3] rzg2: add common configuration header

Most of the values are identical for these 3 boards, so use a common #define
header to hold these common items.

Signed-off-by: Hugo Villeneuve <hvilleneuve@dimonoff.com>
---
 .../{smarc-rzg2l.h => smarc-rzg2-common.h}    | 12 ++--
 include/configs/smarc-rzg2l.h                 | 66 +------------------
 include/configs/smarc-rzg2lc.h                | 66 +------------------
 include/configs/smarc-rzg2ul.h                | 66 +------------------
 4 files changed, 11 insertions(+), 199 deletions(-)
 copy include/configs/{smarc-rzg2l.h => smarc-rzg2-common.h} (86%)

diff --git a/include/configs/smarc-rzg2l.h b/include/configs/smarc-rzg2-common.h
similarity index 86%
copy from include/configs/smarc-rzg2l.h
copy to include/configs/smarc-rzg2-common.h
index b6d2c8af33..b5da03c39a 100644
--- a/include/configs/smarc-rzg2l.h
+++ b/include/configs/smarc-rzg2-common.h
@@ -3,8 +3,8 @@
  * Copyright (C) 2015 Renesas Electronics Corporation
  */
 
-#ifndef __SMARC_RZG2L_H
-#define __SMARC_RZG2L_H
+#ifndef __SMARC_RZG2_COMMON_H
+#define __SMARC_RZG2_COMMON_H
 
 #include <asm/arch/rmobile.h>
 
@@ -43,11 +43,9 @@
 
 #define DRAM_RSV_SIZE			0x08000000
 #define CONFIG_SYS_SDRAM_BASE		(0x40000000 + DRAM_RSV_SIZE)
-#define CONFIG_SYS_SDRAM_SIZE		(0x80000000u - DRAM_RSV_SIZE) //total 2GB
 #define CONFIG_SYS_LOAD_ADDR		0x58000000
 #define CONFIG_LOADADDR			CONFIG_SYS_LOAD_ADDR // Default load address for tfpt,bootp...
 #define CONFIG_VERY_BIG_RAM
-#define CONFIG_MAX_MEM_MAPPED		(0x80000000u - DRAM_RSV_SIZE)
 
 #define CONFIG_SYS_MONITOR_BASE		0x00000000
 #define CONFIG_SYS_MONITOR_LEN		(1 * 1024 * 1024)
@@ -65,8 +63,8 @@
 	"prodsdbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk1p2 \0" \
 	"prodemmcbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
 	"bootimage=unzip 0x4A080000 0x48080000; booti 0x48080000 - 0x48000000 \0" \
-	"emmcload=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a07g044l2-smarc.dtb;run prodemmcbootargs \0" \
-	"sd1load=ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/r9a07g044l2-smarc.dtb;run prodsdbootargs \0" \
+	"emmcload=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/" KERNEL_DT_BASENAME ".dtb;run prodemmcbootargs \0" \
+	"sd1load=ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/" KERNEL_DT_BASENAME ".dtb;run prodsdbootargs \0" \
 	"bootcmd_check=if mmc dev 1; then run sd1load; else run emmcload; fi \0"
 
 #define CONFIG_BOOTCOMMAND	"env default -a;run bootcmd_check;run bootimage"
@@ -75,4 +73,4 @@
 /* Ethernet RAVB */
 #define CONFIG_BITBANGMII_MULTI
 
-#endif /* __SMARC_RZG2L_H */
+#endif /* __SMARC_RZG2_COMMON_H */
diff --git a/include/configs/smarc-rzg2l.h b/include/configs/smarc-rzg2l.h
index b6d2c8af33..19f806a6d4 100644
--- a/include/configs/smarc-rzg2l.h
+++ b/include/configs/smarc-rzg2l.h
@@ -6,73 +6,11 @@
 #ifndef __SMARC_RZG2L_H
 #define __SMARC_RZG2L_H
 
-#include <asm/arch/rmobile.h>
+#define KERNEL_DT_BASENAME "r9a07g044l2-smarc"
 
-#define CONFIG_REMAKE_ELF
+#include "smarc-rzg2-common.h"
 
-#ifdef CONFIG_SPL
-#define CONFIG_SPL_TARGET	"spl/u-boot-spl.scif"
-#endif
-
-/* boot option */
-
-#define CONFIG_CMDLINE_TAG
-#define CONFIG_SETUP_MEMORY_TAGS
-#define CONFIG_INITRD_TAG
-
-/* Generic Interrupt Controller Definitions */
-/* RZ/G2L use GIC-v3 */
-#define CONFIG_GICV3
-#define GICD_BASE	0x11900000
-#define GICR_BASE	0x11960000
-
-/* console */
-#define CONFIG_SYS_CBSIZE		2048
-#define CONFIG_SYS_BARGSIZE		CONFIG_SYS_CBSIZE
-#define CONFIG_SYS_MAXARGS		64
-#define CONFIG_SYS_BAUDRATE_TABLE	{ 115200, 38400 }
-
-/* PHY needs a longer autoneg timeout */
-#define PHY_ANEG_TIMEOUT		20000
-
-/* MEMORY */
-#define CONFIG_SYS_INIT_SP_ADDR		CONFIG_SYS_TEXT_BASE
-
-/* SDHI clock freq */
-#define CONFIG_SH_SDHI_FREQ		133000000
-
-#define DRAM_RSV_SIZE			0x08000000
-#define CONFIG_SYS_SDRAM_BASE		(0x40000000 + DRAM_RSV_SIZE)
 #define CONFIG_SYS_SDRAM_SIZE		(0x80000000u - DRAM_RSV_SIZE) //total 2GB
-#define CONFIG_SYS_LOAD_ADDR		0x58000000
-#define CONFIG_LOADADDR			CONFIG_SYS_LOAD_ADDR // Default load address for tfpt,bootp...
-#define CONFIG_VERY_BIG_RAM
 #define CONFIG_MAX_MEM_MAPPED		(0x80000000u - DRAM_RSV_SIZE)
 
-#define CONFIG_SYS_MONITOR_BASE		0x00000000
-#define CONFIG_SYS_MONITOR_LEN		(1 * 1024 * 1024)
-#define CONFIG_SYS_MALLOC_LEN		(64 * 1024 * 1024)
-#define CONFIG_SYS_BOOTM_LEN		(64 << 20)
-
-/* The HF/QSPI layout permits up to 1 MiB large bootloader blob */
-#define CONFIG_BOARD_SIZE_LIMIT		1048576
-
-/* ENV setting */
-
-#define CONFIG_EXTRA_ENV_SETTINGS \
-	"usb_pgood_delay=2000\0" \
-	"bootm_size=0x10000000 \0" \
-	"prodsdbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk1p2 \0" \
-	"prodemmcbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
-	"bootimage=unzip 0x4A080000 0x48080000; booti 0x48080000 - 0x48000000 \0" \
-	"emmcload=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a07g044l2-smarc.dtb;run prodemmcbootargs \0" \
-	"sd1load=ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/r9a07g044l2-smarc.dtb;run prodsdbootargs \0" \
-	"bootcmd_check=if mmc dev 1; then run sd1load; else run emmcload; fi \0"
-
-#define CONFIG_BOOTCOMMAND	"env default -a;run bootcmd_check;run bootimage"
-
-/* For board */
-/* Ethernet RAVB */
-#define CONFIG_BITBANGMII_MULTI
-
 #endif /* __SMARC_RZG2L_H */
diff --git a/include/configs/smarc-rzg2lc.h b/include/configs/smarc-rzg2lc.h
index ddcae7fc18..1737fd1796 100644
--- a/include/configs/smarc-rzg2lc.h
+++ b/include/configs/smarc-rzg2lc.h
@@ -6,73 +6,11 @@
 #ifndef __SMARC_RZG2L_H
 #define __SMARC_RZG2L_H
 
-#include <asm/arch/rmobile.h>
+#define KERNEL_DT_BASENAME "r9a07g044c2-smarc"
 
-#define CONFIG_REMAKE_ELF
+#include "smarc-rzg2-common.h"
 
-#ifdef CONFIG_SPL
-#define CONFIG_SPL_TARGET	"spl/u-boot-spl.scif"
-#endif
-
-/* boot option */
-
-#define CONFIG_CMDLINE_TAG
-#define CONFIG_SETUP_MEMORY_TAGS
-#define CONFIG_INITRD_TAG
-
-/* Generic Interrupt Controller Definitions */
-/* RZ/G2L use GIC-v3 */
-#define CONFIG_GICV3
-#define GICD_BASE	0x11900000
-#define GICR_BASE	0x11960000
-
-/* console */
-#define CONFIG_SYS_CBSIZE		2048
-#define CONFIG_SYS_BARGSIZE		CONFIG_SYS_CBSIZE
-#define CONFIG_SYS_MAXARGS		64
-#define CONFIG_SYS_BAUDRATE_TABLE	{ 115200, 38400 }
-
-/* PHY needs a longer autoneg timeout */
-#define PHY_ANEG_TIMEOUT		20000
-
-/* MEMORY */
-#define CONFIG_SYS_INIT_SP_ADDR		CONFIG_SYS_TEXT_BASE
-
-/* SDHI clock freq */
-#define CONFIG_SH_SDHI_FREQ		133000000
-
-#define DRAM_RSV_SIZE			0x08000000
-#define CONFIG_SYS_SDRAM_BASE		(0x40000000 + DRAM_RSV_SIZE)
 #define CONFIG_SYS_SDRAM_SIZE		(0x40000000u - DRAM_RSV_SIZE) //total 1GB
-#define CONFIG_SYS_LOAD_ADDR		0x58000000
-#define CONFIG_LOADADDR			CONFIG_SYS_LOAD_ADDR // Default load address for tfpt,bootp...
-#define CONFIG_VERY_BIG_RAM
 #define CONFIG_MAX_MEM_MAPPED		(0x40000000u - DRAM_RSV_SIZE)
 
-#define CONFIG_SYS_MONITOR_BASE		0x00000000
-#define CONFIG_SYS_MONITOR_LEN		(1 * 1024 * 1024)
-#define CONFIG_SYS_MALLOC_LEN		(64 * 1024 * 1024)
-#define CONFIG_SYS_BOOTM_LEN		(64 << 20)
-
-/* The HF/QSPI layout permits up to 1 MiB large bootloader blob */
-#define CONFIG_BOARD_SIZE_LIMIT		1048576
-
-/* ENV setting */
-
-#define CONFIG_EXTRA_ENV_SETTINGS \
-	"usb_pgood_delay=2000\0" \
-	"bootm_size=0x10000000 \0" \
-	"prodsdbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk1p2 \0" \
-	"prodemmcbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
-	"bootimage=unzip 0x4A080000 0x48080000; booti 0x48080000 - 0x48000000 \0" \
-	"emmcload=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a07g044c2-smarc.dtb;run prodemmcbootargs \0" \
-	"sd1load=ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/r9a07g044c2-smarc.dtb;run prodsdbootargs \0" \
-	"bootcmd_check=if mmc dev 1; then run sd1load; else run emmcload; fi \0"
-
-#define CONFIG_BOOTCOMMAND	"env default -a;run bootcmd_check;run bootimage"
-
-/* For board */
-/* Ethernet RAVB */
-#define CONFIG_BITBANGMII_MULTI
-
 #endif /* __SMARC_RZG2L_H */
diff --git a/include/configs/smarc-rzg2ul.h b/include/configs/smarc-rzg2ul.h
index 7a56bc6bf8..affcedcc82 100644
--- a/include/configs/smarc-rzg2ul.h
+++ b/include/configs/smarc-rzg2ul.h
@@ -6,73 +6,11 @@
 #ifndef __SMARC_RZG2UL_H
 #define __SMARC_RZG2UL_H
 
-#include <asm/arch/rmobile.h>
+#define KERNEL_DT_BASENAME "r9a07g043u11-smarc"
 
-#define CONFIG_REMAKE_ELF
+#include "smarc-rzg2-common.h"
 
-#ifdef CONFIG_SPL
-#define CONFIG_SPL_TARGET	"spl/u-boot-spl.scif"
-#endif
-
-/* boot option */
-
-#define CONFIG_CMDLINE_TAG
-#define CONFIG_SETUP_MEMORY_TAGS
-#define CONFIG_INITRD_TAG
-
-/* Generic Interrupt Controller Definitions */
-/* RZ/G2L use GIC-v3 */
-#define CONFIG_GICV3
-#define GICD_BASE	0x11900000
-#define GICR_BASE	0x11960000
-
-/* console */
-#define CONFIG_SYS_CBSIZE		2048
-#define CONFIG_SYS_BARGSIZE		CONFIG_SYS_CBSIZE
-#define CONFIG_SYS_MAXARGS		64
-#define CONFIG_SYS_BAUDRATE_TABLE	{ 115200, 38400 }
-
-/* PHY needs a longer autoneg timeout */
-#define PHY_ANEG_TIMEOUT		20000
-
-/* MEMORY */
-#define CONFIG_SYS_INIT_SP_ADDR		CONFIG_SYS_TEXT_BASE
-
-/* SDHI clock freq */
-#define CONFIG_SH_SDHI_FREQ		133000000
-
-#define DRAM_RSV_SIZE			0x08000000
-#define CONFIG_SYS_SDRAM_BASE		(0x40000000 + DRAM_RSV_SIZE)
 #define CONFIG_SYS_SDRAM_SIZE		(0x40000000u - DRAM_RSV_SIZE) //total 1GB
-#define CONFIG_SYS_LOAD_ADDR		0x58000000
-#define CONFIG_LOADADDR			CONFIG_SYS_LOAD_ADDR // Default load address for tfpt,bootp...
-#define CONFIG_VERY_BIG_RAM
 #define CONFIG_MAX_MEM_MAPPED		(0x40000000u - DRAM_RSV_SIZE)
 
-#define CONFIG_SYS_MONITOR_BASE		0x00000000
-#define CONFIG_SYS_MONITOR_LEN		(1 * 1024 * 1024)
-#define CONFIG_SYS_MALLOC_LEN		(64 * 1024 * 1024)
-#define CONFIG_SYS_BOOTM_LEN		(64 << 20)
-
-/* The HF/QSPI layout permits up to 1 MiB large bootloader blob */
-#define CONFIG_BOARD_SIZE_LIMIT		1048576
-
-/* ENV setting */
-
-#define CONFIG_EXTRA_ENV_SETTINGS \
-	"usb_pgood_delay=2000\0" \
-	"bootm_size=0x10000000 \0" \
-	"prodsdbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk1p2 \0" \
-	"prodemmcbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
-	"bootimage=unzip 0x4A080000 0x48080000; booti 0x48080000 - 0x48000000 \0" \
-	"emmcload=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a07g043u11-smarc.dtb;run prodemmcbootargs \0" \
-	"sd1load=ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/r9a07g043u11-smarc.dtb;run prodsdbootargs \0" \
-	"bootcmd_check=if mmc dev 1; then run sd1load; else run emmcload; fi \0"
-
-#define CONFIG_BOOTCOMMAND	"env default -a;run bootcmd_check;run bootimage"
-
-/* For board */
-/* Ethernet RAVB */
-#define CONFIG_BITBANGMII_MULTI
-
 #endif /* __SMARC_RZG2UL_H */
-- 
2.39.2

